packages/@lwc/engine-core/src/3rdparty/snabbdom/snabbdom.ts

```diff
@@ -171,6 +171,12 @@ export function updateStaticChildren(parentElm: Node, oldCh: VNodes, newCh: VNod
         addVnodes(parentElm, null, newCh, 0, length);
         return;
     }
+    if (length === 0) {
+        // the new list is empty and there's some old nodes, we can directly remove anything old.
+        // this may be the case in which the children are dynamic, and are inside an if.
+        removeVnodes(parentElm, oldCh, 0, oldCh.length);
```

Comment:
Nitpick, I would also dereference `oldCh.length` at the top of the function and rename the current `length` identifier. The `oldCh.length` is now accessed multiple times in the function. 

```ts
const oldLength = oldCh.length;
const newLength = newCh.length;
```